app/assets/javascripts/autocomplete.js

```diff
@@ -130,282 +140,284 @@ var MOAutocompleter = function(opts) {
+    const key = event.which == 0 ? event.keyCode : event.which;
+    // this.debug("keydown(" + key + ")");
     this.clear_key();
     this.focused = true;
     if (this.menu_up) {
       switch (key) {
-      case EVENT_KEY_ESC:
-        this.schedule_hide();
-        this.menu_up = false;
-        break;
-      case EVENT_KEY_RETURN:
-      case EVENT_KEY_TAB:
-        if (this.current_row >= 0)
-          this.select_row(this.current_row - this.scroll_offset);
-        break;
-      case EVENT_KEY_HOME:
-        this.go_home();
-        break;
-      case EVENT_KEY_END:
-        this.go_end();
-        break;
-      case EVENT_KEY_PAGEUP:
-        this.page_up();
-        this.schedule_key(this.page_up);
-        break;
-      case EVENT_KEY_UP:
-        this.arrow_up();
-        this.schedule_key(this.arrow_up);
-        break;
-      case EVENT_KEY_DOWN:
-        this.arrow_down();
-        this.schedule_key(this.arrow_down);
-        break;
-      case EVENT_KEY_PAGEDOWN:
-        this.page_down();
-        this.schedule_key(this.page_down);
-        break;
-      default:
-        this.current_row = -1;
-        break;
+        case EVENT_KEY_ESC:
+          this.schedule_hide();
+          this.menu_up = false;
+          break;
+        case EVENT_KEY_RETURN:
+        case EVENT_KEY_TAB:
+          if (this.current_row >= 0)
+            this.select_row(this.current_row - this.scroll_offset);
+          break;
+        case EVENT_KEY_HOME:
+          this.go_home();
+          break;
+        case EVENT_KEY_END:
+          this.go_end();
+          break;
+        case EVENT_KEY_PAGEUP:
+          this.page_up();
+          this.schedule_key(this.page_up);
+          break;
+        case EVENT_KEY_UP:
+          this.arrow_up();
+          this.schedule_key(this.arrow_up);
+          break;
+        case EVENT_KEY_DOWN:
+          this.arrow_down();
+          this.schedule_key(this.arrow_down);
+          break;
+        case EVENT_KEY_PAGEDOWN:
+          this.page_down();
+          this.schedule_key(this.page_down);
+          break;
+        default:
+          this.current_row = -1;
+          break;
       }
     }
-    if (this.on_keydown)
-      this.on_keydown(event);
     if (this.menu_up && this.is_hot_key(key) &&
-        !(key == EVENT_KEY_TAB || this.current_row < 0))
+      !(key == EVENT_KEY_TAB || this.current_row < 0))
       return false;
     return true;
   },
 
   // Need to prevent these keys from being processed by form.
-  our_keypress: function(event) {
-    var key = event.which == 0 ? event.keyCode : event.which;
-    // jQuery("#log").append("keypress(key=" + key + ", menu_up=" + this.menu_up + ", hot=" + this.is_hot_key(key) + ")<br/>");
-    if (this.on_keypress)
-      this.on_keypress(event);
+  our_keypress: function (event) {
+    const key = event.which == 0 ? event.keyCode : event.which;
+    // this.debug("keypress(key=" + key + ", menu_up=" + this.menu_up + ", hot=" + this.is_hot_key(key) + ")");
```

Comment:
Thanks for keeping these debugging messages!  They were very helpful once upon a time. :)